Packages/com.unity.inputsystem/InputSystem/Editor/AssetImporter/InputActionImporter.cs

```diff
@@ -133,6 +136,53 @@ public override void OnImportAsset(AssetImportContext ctx)
                 GenerateWrapperCode(ctx, asset, m_WrapperCodeNamespace, m_WrapperClassName, m_WrapperCodePath);
         }
 
+        // Case https://jira.unity3d.com/browse/ISXB-1474
+        // Converts any Custom(EnumName=index) processor entries in the JSON to use the enum’s actual underlying value.
+        // Inserts a one-time "__upgradeVersion" field if at least one conversion ran.
+        // Leaves files untouched if there’s nothing to convert or already stamped.
+        static string MigrateJson(string originalJson)
+        {
+            if (originalJson.Contains($"\"{k_VersionFieldName}\""))
+                return originalJson;
+
+            var regex = new Regex("\"processors\"\\s*:\\s*\"Custom\\((\\w+)=(\\d+)\\)\"", RegexOptions.Compiled);
```

Comment:
At least to me this regex is non-trivial. Would you mind adding a comment explaining what conditions are checked for?